android/java/org/chromium/chrome/browser/bookmarks/BraveBookmarkManager.java

```diff
@@ -0,0 +1,151 @@
+import java.io.FileOutputStream;
+import java.io.InputStream;
+
+public class BraveBookmarkManager extends BookmarkManager implements BraveBookmarkDelegate {
+    private ActivityWindowAndroid mWindowAndroid;
+    private BookmarkModel mBookmarkModel;
+    private Context mContext;
+    private static final String TAG = "BraveBookmarkManager";
+
+    public BraveBookmarkManager(Context context, ComponentName openBookmarkComponentName,
+            boolean isDialogUi, boolean isIncognito, SnackbarManager snackbarManager) {
+        super(context, openBookmarkComponentName, isDialogUi, isIncognito, snackbarManager);
+    }
+
+    public void setWindow(ActivityWindowAndroid window) {
+        mWindowAndroid = window;
+    }
+
+    @Override
+    public void importBookmarks() {
+        if (mWindowAndroid == null || mContext == null) {
+            return;
+        }
+
+        if (DeviceFormFactor.isNonMultiDisplayContextOnTablet(mContext)) {
+            doImportBookmarks();
+        } else {
+            if (mWindowAndroid.hasPermission(Manifest.permission.READ_EXTERNAL_STORAGE)) {
+                doImportBookmarks();
+            } else {
+                mWindowAndroid.requestPermissions(
+                        new String[] {Manifest.permission.READ_EXTERNAL_STORAGE},
+                        (permissions, grantResults) -> {
+                            if (grantResults.length >= 1
+                                    && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                                doImportBookmarks();
+                            }
+                        });
+            }
+        }
+    }
+
+    private void doImportBookmarks() {
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.setType("text/html");
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        if (mWindowAndroid.showIntent(
+                    Intent.createChooser(intent, "R.string.import_bookmarks_select_file"),
+                    new WindowAndroid.IntentCallback() {
+                        @Override
+                        public void onIntentCompleted(int resultCode, Intent results) {
+                            if (resultCode == Activity.RESULT_OK && results != null
+                                    && results.getData() != null) {
+                                try {
+                                    Cursor cursor = mContext.getContentResolver().query(
+                                            results.getData(), null, null, null, null);
+                                    int nameIndex =
+                                            cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
+                                    cursor.moveToFirst();
+                                    String name = cursor.getString(nameIndex);
+                                    File file = new File(mContext.getFilesDir(), name);
+                                    InputStream inputStream =
```

Comment:
please use try-with-resources construction as we do in https://github.com/brave/brave-core/blob/master/build/android/bytecode/java/org/brave/bytecode/ByteCodeProcessor.java#L129 to avoid accidental leaks

Reply:
It is already in the try catch block so is it fine not to add try-with-resources construction?

Reply:
idea of try-with-resources construction that it has internal `finally` block and `xxx.close()` method for all acquired resources, so it must be here. 

It is possible to write the same finally block and proper call of close() methods, but try-with-resources is more prefferable.

Reply:
Got it. I'll update the code